models/Events.js

```diff
@@ -212,6 +212,41 @@ class GamePlayEvent extends Event {
     this.floatData = floatData;
     this.booleanData = booleanData;
   }
+  static [Symbol.hasInstance] (obj) {
+    if (obj.type !== "GamePlayEvent") {
+      console.error("wrong event type");
+      return false;
+    }
+    if (obj.senderId === "") {
+      console.error("missing senderId");
+      return false;
+    }
+    if (obj.roomId === "") {
+      console.error("missing room id");
+      return false;
+    }
+    if (obj.eventName === "") {
+      console.error("missing event name");
+      return false;
+    }
+    if (typeof (obj.stringData) === Array) {
+      console.error("stringData is not an array");
```

Comment:
Checks that it "is" an array, but logs that it "is not" an array.

Reply:
Besides that, null is possible. typeData seems quite raw as a generic type. Why not Json object array?

Reply:
Thanks for this, I fixed the conditions and updated the conditions to allow undefined (null) values for stringData, booleanData, intData .
Also added some unit tests to verify that.
36afed8